### PR TITLE
fix 'Html is processing incorrectly on the client side' (close #684)

### LINKED ISSUE #684: Html is processing incorrectly on the client side
```javascript
test('', function () {
    var htmlSrc = [
        '<script>',
        '    for (var b = 0, sum = 0; b <this.i.length; ++b)',
        '        sum += this.i[b] > 5 ? "x" : "y";',
        '<\/script>'
    ].join('\n');

    var htmlExpected = [
        '<script>',
        processScript('\n    for (var b = 0, sum = 0; b <this.i.length; ++b)\n        sum += this.i[b] > 5 ? "x" : "y";\n', true),
        '<\/script>'
    ].join('');
//<script>/*hammerhead|script|start*/if(typeof window!==\"undefined\"&&window){window[\"hammerhead|process-dom-method\"] && window[\"hammerhead|process-dom-method\"]();}else{var __get$Loc=function(l){return l},__set$Loc=function(l,v){return l = v},__set$=function(o,p,v){return o[p] = v},__get$=function(o,p){return o[p]},__call$=function(o,p,a){return o[p].apply(o,a)},__get$Eval=function(e){return e},__proc$Script=function(s){return s},__get$Storage=function(s){return s};}/*hammerhead|script|processing-header-end*/
//
//    for (var b = 0, sum = 0; b < __get$(this.i,\"length\"); ++b)
//         sum=sum+(__get$(this.i,b)>5?\"x\":\"y\");
//
///*hammerhead|script|end*/</script>

    strictEqual(htmlUtils.processHtml(htmlSrc), htmlExpected);
//<script>/*hammerhead|script|start*/if(typeof window!==\"undefined\"&&window){window[\"hammerhead|process-dom-method\"] && window[\"hammerhead|process-dom-method\"]();}else{var __get$Loc=function(l){return l},__set$Loc=function(l,v){return l = v},__set$=function(o,p,v){return o[p] = v},__get$=function(o,p){return o[p]},__call$=function(o,p,a){return o[p].apply(o,a)},__get$Eval=function(e){return e},__proc$Script=function(s){return s},__get$Storage=function(s){return s};}/*hammerhead|script|processing-header-end*/
//
//    for (var b = 0, sum = 0; b < __get$(fake_tag_name_this.i,\"length\"); ++b)
                                          ^^^^^^^^^^^^^^
//         sum=sum+(__get$(this.i,b)>5?\"x\":\"y\");
//
///*hammerhead|script|end*/</script>
});
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
@VasilyStrelyaev check please

File: test/client/fixtures/utils/html-test.js [hunk 1]
```diff
@@ -269,3 +269,18 @@ test('html and body attributes must be processed (T226655)', function () {
        -1);
 });
 
+test('html with special script is processed correctly (GH-684)', function () {
```
Comment (churkin) [inline test/client/fixtures/utils/html-test.js]: "correctly" ?

Comment (VasilyStrelyaev) [inline test/client/fixtures/utils/html-test.js]: yep

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by miherlosev** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Pull request merged**